apps/Gruntfile.js

```diff
@@ -345,12 +345,18 @@ module.exports = function (grunt) {
           ' -p [ factor-bundle -o ' + allFilesDest.join(' -o ') + ' ] -o ' + outputDir + 'common.js' :
           ' -o ' + allFilesDest[0]
       );
+  var applabAPIExec = 'mkdir -p build/browserified && `npm bin`/browserifyinc' +
+      ' --cachefile ' + outputDir + 'applab-api-cache.json' +
+      ' -t [ babelify --compact=false --sourceMap --sourceMapRelative="$PWD" ]' +
+      (envOptions.dev ? '' : ' -t [ envify --NODE_ENV production ]') +
+      ' -d build/js/applab/api-entry.js -o ' + outputDir + 'applab-api.js';
```

Comment:
Here's what I know, though the general feeling on this may have changed since that comment was written - we're [using the JS API over in code-studio now](https://github.com/code-dot-org/code-dot-org/blob/staging/code-studio/scripts/build-commands.js#L43).

* Speed - for a time, we thought browserify's JS API didn't work with streams very well: something browserify+grunt has struggled with in the past and browserify+gulp does much better, but which is all based on the idea of CLI pipes anyway so why not just use CLI?  Streams mean fewer interactions with the filesystem, which can significantly speed up builds.  At this point though, we're not piping anything around - almost everything is a browserify transform, and we're resigned (in both apps and code-studio) to uglify being a separate post-build, post-filesystem-write step, which is really wants to be, so we may not be getting any speedup anymore.
* Stability - I'm not sure what's more stable about the CLI interface, to be honest.  We could ask @wjordan though his thoughts on this may have changed since that comment was written more than a year ago as well.
* Expressiveness and explicitness - There's a part of me that still likes the CLI because it's very compact (compare the ~12 lines to build up this command here vs the ~110 doing nearly the same thing in code-studio) and it's easy to print the exact build command used as part of the build log, which I occasionally find useful. Of course, the 110 lines in code-studio are better documented and easier to modify.  I actually did the transition from one API to the other in code-studio recently, and it's distressing how each form makes certain transformations easier - I can't say for sure that one is better than the other.